tests/benchmark/utils/s2n_negotiate_client_benchmark.cc

```diff
@@ -0,0 +1,231 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+#include <tests/benchmark/utils/s2n_negotiate_client_benchmark.h>
+#include <tests/benchmark/utils/shared_info.h>
+#include <benchmark/benchmark.h>
+
+extern "C" {
+#include "bin/common.h"
+#include <error/s2n_errno.h>
+
+#include "tls/s2n_connection.h"
+
+#include "stuffer/s2n_stuffer.h"
+
+}
+
+
+struct addrinfo hints, *ai_list, *ai;
+int add, sockfd = 0;
+const char *server_name = "localhost";
+const char *cipher_prefs = "test_all_tls12";
+s2n_status_request_type type = S2N_STATUS_REQUEST_NONE;
+char *psk_optarg_list[S2N_MAX_PSK_LIST_LENGTH];
+size_t psk_list_len = 0;
+
+static void setup_config() {
+    memset(&hints, 0, sizeof(hints));
+    hints.ai_family = AF_UNSPEC;
+    hints.ai_socktype = SOCK_STREAM;
+
+    if ((add = getaddrinfo(host, port, &hints, &ai_list)) != 0) {
+        fprintf(stderr, "getaddrinfo error: %s\n", gai_strerror(add));
+        printf("Error: %d\n", add);
+        printf("Errno: %s\n", strerror(errno));
+        exit(1);
```

Comment:
The s2n convention is to return an error to the caller and have the top level figure out what to do, theoretically your benchmark application could retry calling setup but calling exit here prevents that.